### PR TITLE
Adding GREEN LED RX functionality for #232

### PR DESCRIPTION
**Pull request by markwiem**
Created new branch due to suspect Fast-Forward issue in my repo of the original issue232 branch.
Changes need review, but should mirror those of issue232 branch. 

I recommend we close the original issue232 branch: https://github.com/FaradayRF/Faraday-Software/pull/238#pullrequestreview-48692970

### CODE REVIEW

**Comment by kb1lqd** [comment]
@markwiem did you forget one of these two?

https://github.com/FaradayRF/Faraday-Software/pull/238#pullrequestreview-48692970

**Comment by markwiem** [comment]
@kb1lqd I may be missing something, but I feel I've made the required changes. 

From the original PR "bootbitmask[5] should be bootbitmask[2]" is satisfied in line 146 of `simpleconfig.py`

and 

"bootmask[2]. should be bootmask[5]"  is satisfied in line 241 of `deviceconfiguration.py`

Please let me know if I'm still mistaken. Thanks!

File: etc/faraday/faraday_config.sample.ini [hunk 1]
```diff
@@ -1,5 +1,6 @@
 [BASIC]
 REDLEDTX=1
+GREENLEDRX=1
```
Comment (kb1lqd) [inline etc/faraday/faraday_config.sample.ini:3]: Looks good.

File: faraday/deviceconfiguration.py [hunk 2]
```diff
@@ -68,6 +68,8 @@
 parser.add_argument('--nodeid', type=int, help='Set Faraday radio nodeid')
 parser.add_argument('--redledtxon', action='store_true', help='Set Faraday radio RED LED during RF transmissions ON')
 parser.add_argument('--redledtxoff', action='store_true', help='Set Faraday radio RED LED during RF transmissions OFF')
+parser.add_argument('--greenledrxon', action='store_true', help='Set Faraday radio GREEN LED during RF reception ON')
```
Comment (kb1lqd) [inline faraday/deviceconfiguration.py:71]: Good.

File: faraday/deviceconfiguration.py [hunk 3]
```diff
@@ -222,6 +224,10 @@ def configureDeviceConfiguration(args, deviceConfigurationConfigPath, faradayCon
         fconfig.set('BASIC', 'REDLEDTX', 1)
     if args.redledtxoff:
         fconfig.set('BASIC', 'REDLEDTX', 0)
+    if args.greenledrxon:
+        fconfig.set('BASIC', 'GREENLEDRX', 1)
+    if args.greenledrxoff:
+        fconfig.set('BASIC', 'GREENLEDRX', 0)
```
Comment (kb1lqd) [inline faraday/deviceconfiguration.py:230]: Good and works.

File: faraday/deviceconfiguration.py [hunk 4]
```diff
@@ -230,7 +236,9 @@ def configureDeviceConfiguration(args, deviceConfigurationConfigPath, faradayCon
     # Create configuration boot bitmask integer
     bootmask = [0] * 8
     redledtx = fconfig.get('BASIC', 'REDLEDTX')
+    greenledrx = fconfig.get('BASIC', 'GREENLEDRX')
     unitconfigured = fconfig.get('BASIC', 'UNITCONFIGURED')
+    bootmask[5] = greenledrx
```
Comment (kb1lqd) [inline faraday/deviceconfiguration.py:241]: Correct.

File: faraday/simpleconfig.py [hunk 5]
```diff
@@ -143,6 +143,7 @@
 bootbitmask.reverse()
 logger.info("UNITCONFIGURED: {0}".format(bootbitmask[0]))
 logger.info("REDLEDTX: {0}".format(bootbitmask[1]))
+logger.info("GREENLEDRX: {0}".format(bootbitmask[2]))
```
Comment (kb1lqd) [inline faraday/simpleconfig.py:146]: Correct.

**Review by kb1lqd — APPROVED** [review]
I installed this PR and successfully ran it. The unit was programmed with the green led RX enabled and disabled successfully. The configuration read indicates this as did the green LED actuating when expected and visa-verse.

image


Relevant excerpt from the INI file:
```bash
[BASIC]
redledtx = 1
greenledrx = 1
unitconfigured = 1
configbootbitmask = 7
callsign = kb1lqd
id = 2
```

I did not check other functionality but this is considered low-risk per review of the modified code.

Thanks @markwiem !

**Comment by markwiem** [comment]
Awesome! Absolutely made my day! Onward to conquer new mountains! ⛰

**Pull request merged**